hera_cal/utils.py

```diff
@@ -1158,7 +1160,12 @@ def red_average(data, reds=None, bl_tol=1.0, inplace=False,
         nsamples : DataContainer
             If data is a DataContainer, these are its nsamples. Default (None) is 1.0 for all pixels.
             Furthermore, if data is a DataContainer, integration_time is 1.0 for all pixels.
-
+        user_weights_determine_avg_flags : bool, optional
```

Comment:
I think an easier and clearer way to get what you want is to enable this functionality
```
apply_flags_to_wgts : bool, optional
   If True and if wgts is fed, multiply inverse of flags (if provided) into wgts dictionary
```
which does this
```
if apply_flags_to_wgts:
    for k in wgts:
        wgts[k] *= (~flags[k]).atype(np.float)
```
Now you don't have to have two different kinds of output--there is only `favg`, but this `favg` is what you @anguta want when you turn on `apply_flags_to_wgts`. lmk if that makes sense